Build on Mac: clang: error: linker command failed with exit code 1

Hello,
I am trying to build the project on a mac machine, the build reaches 43% then this error popped out: 

```
clang: warning: argument unused during compilation: '-pthread'
ld: unknown option: --as-needed
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [nanocall/list-directory] Error 1
make[1]: *** [nanocall/CMakeFiles/list-directory.dir/all] Error 2
make: *** [all] Error 2

```
Any ideas how to solve this issue?

Thank you for your prompt reply, I removed --as-needed flag and now getting this error: 

Users/xxx/nanocall/src/nanocall/global_assert.hpp:22:16: error: 
      thread-local storage is not supported for the current target
        static thread_local std::string _global_msg;
               ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
2 errors generated.
make[2]: *** [nanocall/CMakeFiles/nanocall.dir/nanocall.cpp.o] Error 1
make[1]: *** [nanocall/CMakeFiles/nanocall.dir/all] Error 2
make: *** [all] Error 2

Looks like Apple's clang does not support thread_local?
I am using 
Apple LLVM version 7.3.0 (clang-703.0.31)
Target: x86_64-apple-darwin15.6.0

Any suggestions are welcome!
Thanks

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -165,7 +165,11 @@ set(CMAKE_CXX_FLAGS_GPROFREL "-O3 -DNDEBUG -DDISABLE_ASSERTS -pg")
 
 # link flags
 #set(CMAKE_EXE_LINKER_FLAGS "-Wl,-rpath=$ENV{LD_RUN_PATH} -Wl,--as-needed")
-set(CMAKE_EXE_LINKER_FLAGS "-Wl,--as-needed")
+if(APPLE)
+	set(CMAKEA_EXE_LINKER_FLAGS "-Wl")
+else()
+	set(CMAKEA_EXE_LINKER_FLAGS "-Wl,--as-needed")
+endif()
 message(STATUS "CMAKE_EXE_LINKER_FLAGS='${CMAKE_EXE_LINKER_FLAGS}'")
 
 ### This target updates the package version

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -166,9 +166,9 @@ set(CMAKE_CXX_FLAGS_GPROFREL "-O3 -DNDEBUG -DDISABLE_ASSERTS -pg")
 # link flags
 #set(CMAKE_EXE_LINKER_FLAGS "-Wl,-rpath=$ENV{LD_RUN_PATH} -Wl,--as-needed")
 if(APPLE)
-	set(CMAKEA_EXE_LINKER_FLAGS "-Wl")
+	set(CMAKE_EXE_LINKER_FLAGS "-Wl")
 else()
-	set(CMAKEA_EXE_LINKER_FLAGS "-Wl,--as-needed")
+	set(CMAKE_EXE_LINKER_FLAGS "-Wl,--as-needed")
 endif()
 message(STATUS "CMAKE_EXE_LINKER_FLAGS='${CMAKE_EXE_LINKER_FLAGS}'")
 

